src/lib/operators/table_scan/single_column_table_scan_impl.cpp

```diff
@@ -61,13 +64,50 @@ void SingleColumnTableScanImpl::handle_value_column(const BaseValueColumn& base_
   });
 }
 
+void SingleColumnTableScanImpl::handle_dictionary_column(const BaseDeprecatedDictionaryColumn& base_column,
+                                                         std::shared_ptr<ColumnVisitableContext> base_context) {
+  _handle_dictionary_column(base_column, base_context);
+}
+
 void SingleColumnTableScanImpl::handle_dictionary_column(const BaseDictionaryColumn& base_column,
                                                          std::shared_ptr<ColumnVisitableContext> base_context) {
+  _handle_dictionary_column(base_column, base_context);
+}
+
+void SingleColumnTableScanImpl::handle_encoded_column(const BaseEncodedColumn& base_column,
+                                                      std::shared_ptr<ColumnVisitableContext> base_context) {
+  auto context = std::static_pointer_cast<Context>(base_context);
+  auto& matches_out = context->_matches_out;
+  const auto& mapped_chunk_offsets = context->_mapped_chunk_offsets;
+  const auto chunk_id = context->_chunk_id;
+
+  const auto left_column_type = _in_table->column_type(_left_column_id);
+
+  resolve_data_type(left_column_type, [&](auto type) {
+    using Type = typename decltype(type)::type;
```

Comment:
I think I have found a solution that works everywhere. Whether it’s a typedef or template variable, what both of them represent is an alias for a type. So one option – which doesn’t collude with already existing types such as `DataType` and `EncodingType` – would be to add the prefix `Alias` to whatever it is. For example in this case, I’d call it `DataTypeAlias`. If it’s referring to any column type, it would be `ColumnAlias`. What do you think?


In case of typedefs that are clearly only referring to a very specific type (a vector or a pointer), we can be precise and just give it a precise name.

Reply:
Unfortunately, "alias" is also used - it could be "SELECT a AS b ...". Especially `ColumnAlias` is problematic (think of `column_expression->alias()`). I feel like we are running out of names. :/

This is not a big issue, as you can simply scroll up and look how Type is defined. Still unhappy about it, but maybe we should leave it for now.